tomcat-embedded-10/src/main/java/org/jboss/arquillian/container/tomcat/embedded/Tomcat10EmbeddedContainer.java

```diff
@@ -0,0 +1,325 @@
+
+        return new ProtocolDescription("Servlet 5.0");
+    }
+
+    @Override
+    public void setup(final TomcatEmbeddedConfiguration configuration) {
+
+        final String serverName = configuration.getServerName();
+
+        if (serverName == null || "".equals(serverName)) {
+            configuration.setServerName("arquillian-tomcat-embedded-10");
+        }
+
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void start() throws LifecycleException {
+
+        try {
+            startTomcatEmbedded();
+        } catch (final Exception e) {
+            throw new LifecycleException("Failed to start embedded Tomcat", e);
+        }
+    }
+
+    @Override
+    public void stop() throws LifecycleException {
+
+        if (wasStarted) {
+            try {
+                stopTomcatEmbedded();
+            } catch (final org.apache.catalina.LifecycleException e) {
+                throw new LifecycleException("Failed to stop Tomcat", e);
+            }
+        }
+    }
+
+    @Override
+    public ProtocolMetaData deploy(final Archive<?> archive) throws DeploymentException {
+
+        try {
+            // Ensure we don't create a corrupted archive by exporting to a file that already exists.
+            deleteWar(archive);
+
+            final File archiveFile = new File(appBase, archive.getName());
+            archive.as(ZipExporter.class).exportTo(archiveFile, true);
+
+            embeddedHostConfig.deployWAR(archive.getName());
+
+            final ContextName contextName = getContextName(archive);
+            final StandardContext standardContext = (StandardContext) host.findChild(contextName.getName());
+            standardContextProducer.set(standardContext);
+
+            final HTTPContext httpContext =
+                new HTTPContext(configuration.getBindAddress(), configuration.getBindHttpPort());
+
+            for (final String mapping : standardContext.findServletMappings()) {
+                httpContext.add(new Servlet(standardContext.findServletMapping(mapping), contextName.getPath()));
+            }
+
+            return new ProtocolMetaData().addContext(httpContext);
+        } catch (final Exception e) {
+            throw new DeploymentException("Failed to deploy " + archive.getName(), e);
+        }
+    }
+
+    @Override
+    public void undeploy(final Archive<?> archive) throws DeploymentException {
+
+        try {
+            embeddedHostConfig.undeployWAR(archive.getName());
+
+            deleteWar(archive);
+        } catch (final Exception e) {
+            throw new DeploymentException("Failed to undeploy " + archive.getName(), e);
+        }
+    }
+
+    @Override
+    public void deploy(final Descriptor descriptor) throws DeploymentException {
+
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public void undeploy(final Descriptor descriptor) throws DeploymentException {
+
```

Comment:
@manovotn I'm not sure why I'm doing these suggestions anymore :) I thought it's a few extra lines, but seems to be programming style. Can you autoformat this somehow? :)

Reply:
Hmm, same formatting is inside embedded tomcat 8 module (and probably elsewhere) - I basically copied that and adjusted as needed with dependency changes and jakarta namespace changes.
I am trying to convince IntelliJ to auto format it, but it it not powerful enough to match your requirement apparently :)
It cannot remove empty first lines in method body. I will do that manually for this one module.

Reply:
Ok, I've pushed the formatting changes and marked your suggestions as resolved. IDE could do something, the rest I had to do manually for this module. Like I said, it isn't my code style, it was mainly copied :)
I don't think it's worth the hassle to format all other modules at this point...

Reply:
That's fine, thanks for cleaning it up a little bit though :)